### PR TITLE
[[ Bug 16796 ]] Ensure deleted object pools are drained on exit.

### PR DESCRIPTION
**Pull request by runrevmark**
On Mac, the termination handler will never return and can occur
within nested wait loops. In this case, all existing deleted
object pools must be drained before quit.

### CODE REVIEW

File: engine/src/object.cpp [hunk 1]
```diff
@@ -5588,10 +5588,13 @@ void MCDeletedObjectsSetup(void)
...
-    // with no parent.
-    MCAssert(MCdeletedobjectpool != nil && MCdeletedobjectpool -> parent == nil);
-    
+	// Teardown occurs in X_close and can occur whilst inside a nested wait.
+	// In particular, on Mac the NSApp willTerminate method will never return
+	// meaning there are inbalanced waits on the stack. Therefore, we must
+	// 'LeaveWait' until the MCdeletedobjetpool has a nil parent.
+	while(MCdeletedobjectpool -> parent != nil)
+		MCDeletedObjectsLeaveWait(true);
+
```
Comment (peter-b) [inline engine/src/object.cpp:5601]: I suggest leaving the `MCAssert(MCdeletedobjectpool != nil)` in ;-)

**Comment by peter-b** [comment]
@livecode-vulcan review ok 0ebccc6

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 0ebccc6b3cdd2eb37afb4de122796b856dd0a198

**Comment by livecode-vulcan** [comment]
😎 test success 0ebccc6b3cdd2eb37afb4de122796b856dd0a198
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2242)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2250)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2235)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2251)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2257)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2263)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2273)

**Pull request merged**